dask/kubernetes/readme.md

```diff
@@ -0,0 +1,72 @@
+# Exploring cuML algorithms with dask-kubernetes in Google Cloud Platform (GCP)
+
+This guide aims to showcase a joint working example of [`cuML`](https://docs.rapids.ai/api/cuml/stable/),
+[`dask-kubernetes`](https://kubernetes.dask.org/en/latest/index.html) and [Google Cloud Platform](https://cloud.google.com/).
+
+## Prerequisite
+
+Install the following package in your environment:
+- [kubectl] (https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/)
+- [gcloud] (https://cloud.google.com/sdk/docs/install)
+
+Login to your gcloud account with
+
+```bash
+gcloud init
+```
+
+## Cluster Setup
+
+First,
+let's setup the cluster to supply the resources required by the examples.
+In total,
+we will use at most 1 dask-scheduler pod and 8 dask-cuda-workers.
+See `spec/sched-spec.yaml` and `spec/worker-spec.yaml` for resource requirements.
+
+```bash
+gcloud container clusters create rapids-dask-kubernetes \
+  --accelerator type=a100,count=8 \
+  --zone us-west1-b \
+  --machine-type a2-highgpu-1g
+```
+
+## Launch Notebook Client
+
+`Dockerfile` contains the image capable of running the notebooks in this folder.
+Build the image:
+
+```bash
+docker build -t rapids-dask-kubernetes-client:22.06 .
+```
+
+Launch the container.
+The container will automatically start a jupyter server and a dask dashboard.
+Expose these services to local ports.
+
+```bash
+docker run --gpus all --rm -it --shm-size=1g --ulimit memlock=-1 -p 8888:8888 -p 8787:8787 -p 8786:8786 -v $HOME/.config/gcloud:/root/.config/gcloud cloud-ml-examples rapids-dask-kubernetes-client:22.06
+```
+
+> **Note**
+> The config files for `gcloud` is bind to the container for reuse,
+> reconfigure if necessary.
+
+Fetch the cluster information in the container:
+
+```bash
+gcloud container clusters get-credentials rapids-dask-kubernetes --region us-west1-b
+```
```

Comment:
Typically this happens automatically when you run `gcloud container clusters create`. Is this not happening for you?

Reply:
The cluster was initialized outside of the jupyter container. So the `kubectl` running inside the container wasn't updated (and wasn't bind to the host's `.kube` folder). Either I can create the cluster inside the container or bind-mount kubectl's config file by some means.

Reply:
So bind mounting `.kube` doesn't work because it seems like `.kube` config files contains information with absolute paths that breaks when mounting it in contianers.

```
Unable to connect to the server: error executing access token command "/home/nfs/wangm/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=fork/exec /home/nfs/wangm/google-cloud-sdk/bin/gcloud: no such file or directory output= stderr=
```

Will adjust the steps to create the cluster inside the container.

Reply:
I'm curious if you mount `/home/nfs/wangm/google-cloud-sdk/` in too whether it would work.

Reply:
It works. Do we want to leave a variable $GCLOUD_ROOT to let the user fill in themselves?